plugin/src/main/groovy/com/novoda/gradle/command/Input.groovy

```diff
@@ -1,80 +1,27 @@
 package com.novoda.gradle.command
 
+import org.gradle.api.Action
 import org.gradle.api.tasks.TaskAction
 
 class Input extends AdbTask {
 
-    Closure script
+    InputSpec script = new InputSpec()
 
-    void text(String value) {
-        input('text', "$value")
-    }
-
-    void tap(int x, int y) {
-        input('touchscreen', 'tap', x, y)
-    }
-
-    void swipe(int startX, int startY, int endX, int endY) {
-        input('touchscreen', 'swipe', startX, startY, endX, endY)
-    }
-
-    void key(int code) {
-        input('keyevent', code)
-    }
-    
-    void home() {
-        key 3
-    }
-    
-    void back() {
-        key 4
-    }
-
-    void up() {
-        key 19
-    }
-
-    void down() {
-        key 20
-    }
-    
-    void left() {
-        key 21
-    }
-
-    void right() {
-        key 22
-    }
-
-    void power() {
-        key 26
-    }
-    
-    void clear() {
-        key 28
-    }
-
-    void tab() {
-        key 61
-    }
-    
-    void enter() {
-        key 66
-    }
-
-    void unlock() {
-        key 82
-    }
-
-    private input(... values) {
-        def command = ["shell", "input"]
-        command.addAll(values)
-        runCommand(command)
+    @Deprecated
```

Comment:
You're talking from the perspective of the sample we provide, not the real public API: if you remove public methods you're introducing breaking changes, right? Duplication in this case is not correct, because you will not duplicate the implementation, will just be tedious to make those method to use delegation instead. I feel we can bare with if we have a proper strategy to move forward :)